src/main/kotlin/com/github/imflog/schema/registry/Subject.kt

```diff
@@ -9,6 +13,10 @@ data class Subject(
 ) {
     val references: MutableList<SchemaReference> = mutableListOf()
     val localReferences: MutableList<LocalReference> = mutableListOf()
+    var metadata: Metadata = Metadata(mutableMapOf(), mutableMapOf(), mutableSetOf())
+    var ruleSet: RuleSet = RuleSet(mutableListOf(), mutableListOf())
```

Comment:
I think the default value should be null values.
Like this we would not register anything in the rules / metadata if nothing is provided.